src/state.rs

```diff
@@ -97,14 +97,16 @@ impl State {
     let (width, height) = (image.width() as usize, image.height() as usize);
 
     self.matrix = DMatrix::from_iterator(
-      height,
       width,
+      height,
       image
         .rows()
         .map(|row| row.map(|pixel| Vector3::new(pixel[0], pixel[1], pixel[2])))
         .flatten(),
     );
 
+    self.matrix = self.matrix.transpose();
```

Comment:
```suggestion
    self.matrix = DMatrix::from_iterator(
      width,
      height,
      image
        .rows()
        .map(|row| row.map(|pixel| Vector3::new(pixel[0], pixel[1], pixel[2])))
        .flatten(),
    ).transpose();
```

A bit cleaner? TIOLI